### PR TITLE
[SCP-1752] feat: gsc sdk add x source header in the configuration

### PR DESCRIPTION
**Pull request by miguel-serfe**
Coverage:
feat/scp-1752-gsc-sdk-add-x-source-header-in-the-configuration

```
  Classes: 95.08% (116/122)  
  Methods: 98.19% (544/554)  
  Lines:   94.06% (3278/3485)
```
gsc-master:

```
  Classes: 95.08% (116/122)  
  Methods: 98.22% (551/561)  
  Lines:   94.34% (3236/3430)
```

### CODE REVIEW

File: src/Application/Configuration.php [hunk 1]
```diff
@@ -26,12 +26,18 @@ class Configuration
...
+     */
+    private $source;
+
+    public function __construct(string $key, string $username, string $endpoint, string $version = '1.0', ?string $source = 'SDK')
     {
         $this->key = $key;
         $this->username = $username;
         $this->endpoint = $endpoint;
         $this->version = $version ?? '1.0';
+        $this->source = $source ?? 'SDK';
```
Comment (otzi122) [inline src/Application/Configuration.php:40]: Isn't it redundant if you are already initializing the source in the method parameters?

**Review by otzi122 — APPROVED** [review]

File: src/Application/Configuration.php (hunk 1 above)
Comment (miguel-serfe) [reply src/Application/Configuration.php]: Fixed. Thanks!

**Review by otzi122 — APPROVED** [review]

**Review by lhbarrientosz — APPROVED** [review]

**Review by sebastiane — APPROVED** [review]

File: src/Application/Configuration.php [hunk 2]
```diff
@@ -53,4 +63,9 @@ public function getVersion(): string
     {
         return $this->version;
     }
+
+    public function getSource(): ?string
```
Comment (yinosanchez) [inline src/Application/Configuration.php:67]: I don't see this method being called anywhere. I think you should remove it

Comment (miguel-serfe) [reply src/Application/Configuration.php:67]: Hi Marcelo, this method is used in the BaseManager inside the method generateRequestHeaders, to set the headers:
https://github.com/LinioIT/seller-center-sdk/pull/94/files#diff-d1a09b8202dddee91bcfa84e6dcc7960291dcd27b8458092a3402ce3f547da32R80

**Review by maruamallo — APPROVED** [review]

**Pull request merged**